astropy/coordinates/representation.py

```diff
@@ -42,8 +42,12 @@ def __init__(cls, name, bases, dct):
         if cls.__name__ == 'BaseRepresentation':
             return
 
-        REPRESENTATION_CLASSES[cls.get_name()] = cls
+        repr_name = cls.get_name()
 
+        if repr_name in REPRESENTATION_CLASSES:
+            raise ValueError("Representation class {0} already defined".format(repr_name))
+
+        REPRESENTATION_CLASSES[repr_name] = cls
```

Comment:
I think that's just fine.  It's sort of implicitly enforced by the existence of `REPRESENTATION_CLASSES`, anyway.  Probably better to *explicitly* require this.  

The one downside is that it prevents, say, user-level code from replacing `SphericalCoordinates` with a custom class.  But that's probably *good*, as it might have unintended consequences.  Better to force them to call it `FancySphericalCoordinates`/`'fancyspherical'` or something